libraries/Microsoft.Bot.Builder/Adapters/TestAdapter.cs

```diff
@@ -172,7 +172,12 @@ public async Task ProcessActivityAsync(Activity activity, BotCallbackHandler cal
 
             if (activity.Timestamp == null || activity.Timestamp == default(DateTimeOffset))
             {
-                activity.Timestamp = DateTime.UtcNow;
+                activity.Timestamp = DateTimeOffset.UtcNow;
+            }
+
+            if (activity.LocalTimestamp == null || activity.LocalTimestamp == default(DateTimeOffset))
+            {
+                activity.LocalTimestamp = DateTimeOffset.Now;
```

Comment:
What was this fixing?

Reply:
time recognizer tests would fail when run in the morning because it was executing in UTC, so tommorrow gives you a different date.  TestAdapter should have been setting this.

---
In reply to: [367263801](https://github.com/microsoft/botbuilder-dotnet/pull/3253#discussion_r367263801) [](ancestors = 367263801)